hera_cal/io.py

```diff
@@ -852,6 +852,89 @@ def partial_time_io(hd, times, **kwargs):
     return combined_hd.build_datacontainers()
 
 
+def save_redcal_meta(meta_filename, fc_meta, omni_meta, freqs, times, lsts, antpos, history):
+    '''Saves redcal metadata to a hdf5 file. See also read_redcal_meta.
+
+    Arguments:
+        meta_filename: path to hdf5 file to save
+        fc_meta: firstcal metadata dictionary, such as that produced by redcal.redcal_iteration()
+        omni_meta: omnical metadata dictionary, such as that produced by redcal.redcal_iteration()
+        freqs: 1D numpy array of frequencies in the data
+        times: 1D numpy array of times in the data
+        lsts: 1D numpy array of LSTs in the data
+        antpos: dictionary of antenna positions in the form {ant_index: np.array([x,y,z])}
+        history: string describing the creation of this file
+    '''
+    with h5py.File(meta_filename, "w") as outfile:
+        # save the metadata of the metadata
+        header = outfile.create_group('header')
+        header['freqs'] = freqs
+        header['times'] = times
+        header['lsts'] = lsts
+        antnums = np.array(sorted(list(antpos.keys())))
+        header['antpos'] = np.array([antpos[antnum] for antnum in antnums])
+        header['antpos'].attrs['antnums'] = antnums
+        header['history'] = history.encode('utf8')
```

Comment:
It's generally safer to cast `str` types to numpy strings using the `np.string_` method. So something like:
```
header['history'] = np.string_(history)
```
Without special handling, h5py behind the scenes will make an "object"-type dataset for `str` types (not sure what happens if you encode it first, as is done here), rather than a "string"-type one, which makes them more broadly compatible outside of python.

Reply:
Fixed.